libtenzir/builtins/functions/hex.cpp

```diff
@@ -0,0 +1,92 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "tenzir/arrow_utils.hpp"
+#include "tenzir/tql2/plugin.hpp"
+
+#include <boost/algorithm/hex.hpp>
+
+namespace tenzir::plugins::hex {
+
+namespace {
+
+enum class mode { encode, decode };
+
+template <mode Mode>
+class plugin final : public function_plugin {
+  using Type = std::conditional_t<Mode == mode::encode, string_type, blob_type>;
```

Comment:
```suggestion
  using coding_type = std::conditional_t<Mode == mode::encode, string_type, blob_type>;
```

We usually have members, including types, snake_case.

Reply:
Not for dependent types - https://github.com/tenzir/tenzir/blob/main/libtenzir%2Fbuiltins%2Ffunctions%2Fint.cpp#L22-L25

Reply:
Interesting, that's new to me.

Reply:
I've unresolved this, as the instance you've pointed out seems to be a one-off. I can find many other dependent types in our code base that are spelled in `snake_case`, and that's also what's documented in the style guide.

Reply:
I think we should follow the template typename style here.

Reply:
It was probably me who introduced this, my bad. You should find it in a bunch of other places too. 

(To be fair, there is not a big difference between template arguments and (private) types derived from them, so using different cases here seems useless, and you know my opinion on the naming scheme in general. But we can go all lower-case for all non-template-parameter-types, sure, I'll stick to it.)

Reply:
I have changed it to match the documented style guide for now. We can come back to this after we have asked for a vote.